implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java

```diff
@@ -142,6 +142,14 @@ private ConfigSourceInterceptorContext buildInterceptorChain(final SmallRyeConfi
         SmallRyeConfigSourceInterceptorContext current = new SmallRyeConfigSourceInterceptorContext(
                 (ConfigSourceInterceptor) (context, name) -> {
                     for (ConfigSource configSource : getConfigSources()) {
+                        if (configSource instanceof ConfigValueConfigSource) {
+                            ConfigValueConfigSource configValueConfigSource = (ConfigValueConfigSource) configSource;
+                            ConfigValue value = configValueConfigSource.getConfigValue(name);
+                            if (value != null) {
+                                return value;
+                            }
+                        }
+
                         String value = configSource.getValue(name);
```

Comment:
If `configSource` is a `ConfigValueConfigSource` then we should never call `getValue(String)` on it.  IOW, this should be in an `else`.